pandas/core/dtypes/cast.py

```diff
@@ -393,7 +398,7 @@ def maybe_promote(dtype, fill_value=np.nan):
             dtype = np.float64
             fill_value = np.nan
         elif is_datetime_or_timedelta_dtype(dtype):
```

Comment:
would it be better to branch this to be explict (e.g. datetime and timedelta branches)

Reply:
actually why do we have a branch for a subclass of np.datetime64 AND this one (and same for timedelta)?

Reply:
not clear.  my current plan is to do a couple more small PRs of xfail-fixing before getting into refactoring